packages/ember-metal/lib/property_get.ts

```diff
@@ -97,6 +94,16 @@ export function get(obj: object, keyName: string): any {
     typeof keyName !== 'string' || keyName.lastIndexOf('this.', 0) !== 0
   );
   assert('Cannot call `get` with an empty string', keyName !== '');
+  if (
+    (obj as ExtendedObject).isDestroyed &&
+    keyName !== 'isDestroyed' &&
+    keyName !== 'isDestroying'
+  ) {
+    warn(`calling \`get\` on destroyed object: ${toString(obj)}.${keyName}`, false, {
+      id: 'calling-get-on-destroyed',
+    });
+    return;
```

Comment:
In general, I think a warning when accessing properties on destroyed objects is probably fine, but I do not think we should prevent accessing the underlying state (at least not right away).

Can you refactor this so that the conditionals are all part of the `warn` call (to ensure that it is all removed properly from production builds) and remove the early return?

Reply:
so I should also remove `return` so it doesn't prevent accessing property of destroyed object ?

Reply:
yes, and remove the conditional (moving those conditions to the `warn` itself)

Reply:
done, but `warn` seems to not lazily evaluate string part of warn

Reply:
added PR to `babel-plugin-debug-macros` https://github.com/chadhietala/babel-plugin-debug-macros/pull/72 to support lazy evaluation of string part of warn